lib/engine/game/g_18_eus/game.rb

```diff
@@ -333,6 +334,22 @@ def apply_subsidy(corporation)
         def remove_subsidy(hex_id)
           hex_by_id(hex_id).tile.icons.reject! { |icon| icon.name.include?('subsidy') }
         end
+
+        def setup_privates
+          @companies.sort_by! { rand }
+          privates = @companies.group_by { |p| p.id[0] }
+          privates.each do | group, comps |
+            first = comps.find { |c| c.id == group + "0" }
+            comps.delete(first)
+            comps.unshift(first)
+          end
+
+          @companies.clear
+          @companies.concat privates["A"].first(4)
+          @companies.concat privates["B"].first(4)
+          @companies.concat privates["C"].first(4)
```

Comment:
@companies = privates.map {|g| g.first(4)}.flatten

Reply:
Will that guarantee that they are in order "As Bs Cs"?

Reply:
You may have to sort the groups first. Alternatively you could do %w[A B C].map...

Reply:
@companies = privates.values.sort.map {|v| v.first(4)}.flatten works

Reply:
A little by luck that the internal implementation creates the group_by map keys in the order you want. Not necessarily something you can count on going forward.